IHP/ModelSupport.hs

```diff
@@ -566,21 +566,66 @@ class
     --
     columnNames :: [ByteString]
 
-    -- | Returns WHERE conditions to match an entity by it's primary key
+    -- | Returns the list of column names, that are contained in the primary key for a given model
+    --
+    -- __Example:__
+    --
+    -- >>> primaryKeyColumnNames @User
+    -- ["id"]
+    --
+    -- >>> primaryKeyColumnNames @PostTagging
+    -- ["post_id", "tag_id"]
+    --
+    primaryKeyColumnNames :: [ByteString]
+
+    -- | Returns WHERE conditions to match an entity by it's primary key, given the entities id
     --
     -- For tables with a simple primary key this returns a tuple with the id:
     --
-    -- >>> primaryKeyCondition project
+    -- >>> primaryKeyConditionForId project.id
     -- [("id", "d619f3cf-f355-4614-8a4c-e9ea4f301e39")]
     --
     -- If the table has a composite primary key, this returns multiple elements:
     --
-    -- >>> primaryKeyCondition postTag
+    -- >>> primaryKeyConditionForId postTag.id
     -- [("post_id", "0ace9270-568f-4188-b237-3789aa520588"), ("tag_id", "0b58fdf5-4bbb-4e57-a5b7-aa1c57148e1c")]
-    --
-    primaryKeyCondition :: record -> [(Text, PG.Action)]
-    default primaryKeyCondition :: forall id. (HasField "id" record id, ToField id) => record -> [(Text, PG.Action)]
-    primaryKeyCondition record = [("id", toField record.id)]
+    primaryKeyConditionForId :: Id record -> [(Text, PG.Action)]
```

Comment:
I can't find any place within the IHP repo where the first field of the returned tuples of `primaryKeyConditionForId` is actually used, and arguably it really is redundant with the existence of `primaryKeyColumnNames`

I wonder if it would be a good idea to convert `primaryKeyConditionForId` and `primaryKeyCondition` to just return `[PG.Action]`

Or we could retain backward compatibility by just converting `primaryKeyConditionForId` to return just the actions and for `primaryKeyCondition` we zip `primaryKeyColumnNames` together with the actions...

I'm not sure what that would buy us tho, it just feels a bit weird to me to have this redundant information...

Reply:
> I wonder if it would be a good idea to convert primaryKeyConditionForId and primaryKeyCondition to just return [PG.Action]

👍

Reply:
Looking at the git blame on when `primaryKeyCondition` was introduced, it looks it was mainly used by `deleteRecord`. So b.c. isn't much of a concern here.

Reply:
Alrighty